Use yarn and resolve dependency conflicts

ebma opened the pull request:
I forgot to change this repository to yarn as well.
Related to https://github.com/pendulum-chain/tasks/issues/402.

ebma committed:
Use yarn and resolve dependency conflicts

ebma committed:
Adjust mentions of npm with yarn

gianfra-t reviewed (approved):
Thanks @ebma. I actually changed it yesterday but later rolled back since on the `README` it was stated that "`yarn` package manager is not supported". Was that a leftover comment? Or it was still a recommendation?

ebma commented:
Yeah good point, I removed that as part of this PR. It was part of the original template that the subsquid team created for substrate squids. I think it's because some of their commands explicitly used `npm` but I changed it so that everything should work with yarn. I don't see why it wouldn't work anymore so I would consider it an outdated comment from the past.
